kernel_gateway/gatewayapp.py

```diff
@@ -225,8 +225,12 @@ def init_configurables(self):
                 raise RuntimeError('cannot prespawn {}; more than max kernels {}'.format(
                     self.prespawn_count, self.max_kernels)
                 )
-            for _ in range(self.prespawn_count):
-                self.kernel_manager.start_kernel()
+        else:
```

Comment:
If I mange to set api to 'foo_bar' the execution is going to go into this else. Is that what we want?